packages/app/src/cli/models/app/loader.test.ts

```diff
@@ -758,6 +758,617 @@ automatically_update_urls_on_dev = true
     expect(myFunction.entrySourceFilePath).toContain(joinPath(blockPath('my-function'), 'src', 'index.js'))
   })
 
+  test('loads the app with a Flow trigger extension that has a full valid configuration', async () => {
+    // Given
+    await writeConfig(appConfiguration)
+
+    const blockConfiguration = `
+      [[extensions]]
+      name = "Auction bid placed"
+      description = "An auction bid has been placed"
+      type = "flow_trigger"
+      handle = "handle1"
+
+      [settings]
+
+        [[settings.fields]]
+        type = "customer_reference"
+
+        [[settings.fields]]
+        type = "single_line_text_field"
+        key = "your_field_key"
```

Comment:
nit: the `_` are technically invalid for the field `key` property. Only alphanumeric chars and spaces (a-Z, 0-9). Also this validation is only for triggers fields `key` property ATM which can be confusing. Actions field `key` can have non alphanumeric characters in it.

Here's the most up to date template on `main`:

```
[[extensions]]
type = "flow_trigger"
name = "{{ name }}"
handle = "{{ handle }}"
description = "Your description"

[settings]

  [[settings.fields]]
  type = "customer_reference"

  [[settings.fields]]
  type = "single_line_text_field"
  key = "Your field key"
```

Reply:
For a metafield definition, key must match:

* regex: /\A[a-zA-Z0-9_-]+\z/ (https://github.com/Shopify/shopify/blob/main/components/content/metafields/app/models/metafield_definition.rb#L20)
* length:  min 3, max 64 (https://github.com/Shopify/shopify/blob/main/components/content/metafields/app/models/metafield_definition.rb#L45)
* must be unique under (shop_id, owner_type, owner_id, namespace)

Reply:
I believe unstructured Metafields (w/o definitions) only have the length and uniqueness constraints.